climada/engine/impact.py

```diff
@@ -390,6 +388,47 @@ def impact_per_year(self, all_years=True, year_range=None):
             year_set[year] = sum(self.at_event[orig_year == year])
         return year_set
 
+    def impact_at_reg(self, agg_regions=None):
+        """Aggregate impact on given aggregation regions. This method works
+        only if Impact.imp_mat was stored during the impact calculation.
+
+        Parameters
+        ----------
+        agg_regions : np.array, list, pd.Series (optional)
+            It has a lenght equal to the number of coordinates points in exposure
+            and it reports what macro-regions these points belong to. For example,
+            asuming there are three centroids and agg_regions = ['A', 'A', 'B']
+            then impact of the first and second centroids will be assigned to
+            region A, whereas impact from the second centroid will be assigned
+            to area B. If no aggregation regions are passed, the method aggregates
+            impact at the country (admin_0) level.
+            Default is None.
+
+        Returns
+        -------
+        pd.DataFrame
+            Contains the aggregated data per event.
+            Rows: Hazard events. Columns: Aggregation regions.
+        """
+        if self.imp_mat.nnz == 0:
+            raise ValueError("The aggregated impact cannot be computed as no"
+                             "Impact.imp_mat was stored during the impact calculation")
+
+        if agg_regions is None:
+            agg_regions = pd.Series(u_coord.get_country_code(self.coord_exp[:,0],
+                                                             self.coord_exp[:,1]))
```

Comment:
Maybe using the alpha-3 codes instead of the numeric codes would be more descriptive?

```suggestion
            agg_regions = pd.Series(
                u_coord.country_to_iso(
                    u_coord.get_country_code(self.coord_exp[:,0], self.coord_exp[:,1])
                )
            )
```

Reply:
Implemented.

Reply:
Excellent!